src/solver/application/application.cpp

```diff
@@ -113,45 +95,49 @@ void Application::prepare(int argc, char* argv[])
         pStudy = nullptr;
         return;
     }
+}
 
-    // Perform some checks
-    checkAndCorrectSettingsAndOptions(pSettings, options);
-
-    pSettings.checkAndSetStudyFolder(options.studyFolder);
-
-    checkStudyVersion(pSettings.studyFolder);
-
-    // Determine the log filename to use for this simulation
-    resetLogFilename();
-
+void Application::startSimulation(Data::StudyLoadOptions options)
+{
     // Starting !
-#ifdef GIT_SHA1_SHORT_STRING
+    #ifdef GIT_SHA1_SHORT_STRING
     logs.checkpoint() << "Antares Solver v" << ANTARES_VERSION_STR << " (" << GIT_SHA1_SHORT_STRING
                       << ")";
-#else
+    #else
     logs.checkpoint() << "Antares Solver v" << ANTARES_VERSION_STR;
-#endif
+    #endif
     WriteHostInfoIntoLogs();
 
-    // Write command-line options into logs
-    // Incidentally, it also seems to contain the full path to the executable
-    logs.info();
-    WriteCommandLineIntoLogs(argc, argv);
+    WriteCommandLineIntoLogs(pArgc, pArgv);
 
     logs.info() << "  :: log filename: " << logs.logfile();
-    // Temporary use a callback to count the number of errors and warnings
+
     logs.callback.connect(this, &Application::onLogMessage);
 
-    // Allocate a study
     pStudy = std::make_shared<Antares::Data::Study>(true /* for the solver */);
 
-    // Setting global variables for backward compatibility
     pParameters = &(pStudy->parameters);
 
-    // Loading the study
     readDataForTheStudy(options);
 
-    // Some more checks require the existence of pParameters, hence of a study.
+    postParametersChecks();
+
+    pStudy->initializeProgressMeter(pSettings.tsGeneratorsOnly);
+    if (pSettings.noOutput)
+        pSettings.displayProgression = false;
+
+    if (pSettings.displayProgression)
```

Comment:
```suggestion
    if (pSettings.noOutput)
        pSettings.displayProgression = false;

    if (pSettings.displayProgression =!pSettings.noOutput; pSettings.displayProgression)
```

Reply:
I will leave it as it is. I want to introduce minimal change of code in this PR and try to keep it as simply encapsulating code in new function